DirectOutput/LedControl/Loader/TableConfigSetting.cs

```diff
@@ -745,10 +745,16 @@ public void ParseSettingData(string SettingData)
                     }
                     IntegerCnt++;
                 }
-                else if (PartNr == 0)
+                // if Parts[PartNr] starts with capital letter and the rest small caps letters or underscore
+                else if (Parts[PartNr].Length > 2 && char.IsUpper(Parts[PartNr][0]) && Parts[PartNr].Skip(1).All(c => (char.IsLower(c) || c == '_')))
                 {
-                    //This should be a color
-                    ColorName = Parts[PartNr];
+                    // This should be a color
+                    ColorName = Parts[PartNr].ToUpper();
+                }
+                else if (Parts[PartNr].Length >= 7 && Parts[PartNr].Length <= 9 && Parts[PartNr][0] == '#' && Parts[PartNr].Substring(1).All(c => Uri.IsHexDigit(c)) )
+                {

```

Comment:
This would also be improved by changing to a regex: `if (Regex.IsMatch(Parts[PartNr], @""^#[0-9A-Fa-f]{6,8)$))`.  (I'm not actually sure that we should enforce the format so strictly, though.  It might be better diagnostic-wise to make this simply `if (Regex.IsMatch(Parts[PartNr], "@^#"))`, since it doesn't seem that a string starting with '#' could be anything other than an html-style color code; and then, only after matching that, add a second test against the full regex, and log an error if it doesn't match ("Invalid '#' color code format, #rrggbb or #rrggbbaa required").